doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,852 @@
+
+## 1.3 Design Overview
+
+### 1.3.1 Container
+The existing "macsec" docker holds all the port security applications. Code changes are also made to the SWSS docker.
+
+### 1.3.2 SAI Support
+No changes to SAI spec for supporting PAC.
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+
+The following figure illustrates how clients like PCs and printers are authenticated and authorized for accessing the network.   
+
+![pac-deployment](images/PAC_Deployment.JPG)
+
+
+**Figure 1 : PAC target deployment use cases**   
+
+## 2.2 Functional Description
+
+PAC uses authentication methods 802.1x and MAB for client authentication. These methods in turn use RADIUS for client credential verification and receive the authorization attributes like VLANs, for the authenticated clients. 
+
+### 2.2.1 802.1x
+
+PAC leverages the IEEE 802.1X-2004 for 802.1x standard as available in the "hostapd" implementation in the "macsec" docker. It is an IEEE Standard for Port Access Control that provides an authentication mechanism to devices wishing to attach to a LAN. The standard defines Extensible Authentication Protocol over LAN (EAPoL), which is an encapsulation technique to carry EAP packets between the Supplicant and the Authenticator. The standard describes an architectural framework within which authentication and consequent actions take place. It also establishes the requirements for a protocol between the Authenticator and the Supplicant, as well as between the Authenticator and the Authentication server. 
+
+### 2.2.2 MAC Authentication Bypass
+
+PAC makes use of MAC Authentication Bypass (MAB) feature to authenticate devices like cameras or printers which do not support 802.1x. MAB makes use of the device MAC address to authenticate the client.
+
+### 2.2.3 RADIUS 
+***Authentication***
+
+PAC (Authenticator) uses an external RADIUS server for client authentication. It determines the authorization status of the clients based on RADIUS Access-Accept or Access-Reject frames as per the RADIUS RFC 2865. 
+
+PAC as a PAE Authenticator for 802.1x is essentially a passthrough for client Authentication exchange messages. Hence different EAP authentication methods like EAP-MD5, EAP-PEAP, EAP-TLS, etc. are supported. These are essentially the 802.1x Supplicant and RADIUS server functionalities.
+
+PAC as a PAE Authenticator for MAB mimics the Supplicant role for MAB clients. Authentication methods EAP-MD5, EAP-PAP and EAP-CHAP are supported.
+
+***Authorization***
+
+Once a client is authenticated, authorization parameters from RADIUS can be sent for the client. The Authenticator switch processes these RADIUS attributes to apply to the client session. Following attributes are supported.
+
+- *VLAN Id*: This is the VLAN ID sent by a RADIUS server for the authenticated client. This VLAN should be a pre-created VLAN on the switch. 
+- *Session Timeout*: This is the timeout attribute of the authenticated client session.
```

Comment:
Do we have default session for un-authenticated clients as well? 

For instance lets say if a MAB  auth-request is rejected by Radius server and if we dont have a session timeout and immediately remove the client. Then the next packet from the client would trigger another Radius request, incase if the client is generating too many packets, the Radius connection might be overloaded. 

One way to maintain a session timer for auth-failed clients as well. Please let me know if this was considered during the design